src/ui/Factory.cpp

```diff
@@ -90,7 +90,7 @@ std::shared_ptr<Step> Factory::createStep(const std::string &command) {
 }
 
 std::shared_ptr<Machine> Factory::createMachine(const State &state) {
-    return std::make_shared<Machine>(std::shared_ptr<ModelInterface>(new TaskManager), shared_from_this(), state);
+    return std::make_shared<Machine>(nullptr,shared_from_this(), state);
```

Comment:
what is this?

Reply:
should've been nullptr all along - inner submachine does not need a model, i.e. unnecessary TaskManager dependency